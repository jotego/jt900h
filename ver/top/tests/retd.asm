    ; call [cc],mem
    main section code
    org 0
    ld a,0xbf    ; common header

    ld xsp,stack
    ld wa,0xcafe
    push wa
    ld wa,0xbeef
    push wa
    call mymove
    ld wa,0xbabe
    cp xsp,stack
    jp ne,bad_end
    jp end_loop

mymove:
    ld bc,(xsp+4)
    ld de,(xsp+6)
    retd 4
    jp bad_end

end_loop:
    ldf 0
    ld hl,0xbabe
    ld (0xffff),0xff
    jp end_loop
bad_end:
    ld hl,0xdead
    ld (0xffff),0xff
    jp bad_end
data:
    dw 0x0011,0x2233,0x4455,0x6677,0x8899,0x0000,0x0000,0x0000
    dw 0x0000,0x0000,0x0000,0x0000,0x0000,0x0000,0x0000,0x0000
copy:
    dw 0x0000,0x0000,0x0000,0x0000,0x0000,0x0000,0x0000,0x0000
stack:
    end