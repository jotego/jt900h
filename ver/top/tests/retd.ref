0000babe - 00000000 - 00000000 - 00000000
0000beef - 00000000 - 00000000 - 00000000
0000cafe - 00000000 - 00000000 - 00000000
0000babe - 00000000 - 00000000 - 00000000
-- index registers --
00000000 - 00000000 - 00000000 - 0000007c
SR = f842
