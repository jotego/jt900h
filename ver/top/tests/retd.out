0000babe - 00000000 - 00000000 - 00000000
0000beef - 00000000 - 00000000 - 00000000
0000cafe - 00000000 - 00000000 - 00000000
0000babe - 00000000 - 00000000 - 00000000
-- index registers --
00000000 - 00000000 - 00000000 - 00000072
SR = 42f8
